### ISSUE TITLE
The estimated cost for `mac1.metal` AWS instance type returns $0

### ISSUE THREAD

**Issue by vdmgolub** [infracost/infracost#2103]
A user reported an issue with the `mac1.metal` instance type:

> I ran an estimated for as Instance usage (Linux/UNIX, on-demand, mac1.metal) and it came back as $0.00

Debug log:
> No products found for aws_instance.web_app Instance usage (Linux/UNIX, on-demand, mac1.metal), using 0.00

TF code to reproduce:
```tf
provider "aws" {
  region                      = "us-east-1" # <<<<< Try changing this to eu-west-1 to compare the costs
  skip_credentials_validation = true
  skip_requesting_account_id  = true
  access_key                  = "mock_access_key"
  secret_key                  = "mock_secret_key"
}

resource "aws_instance" "web_app" {
  ami           = "ami-674cbc1e"
  instance_type = "mac1.metal" # <<<<< Try changing this to m5.8xlarge to compare the costs

  root_block_device {
    volume_size = 50
  }

  ebs_block_device {
    device_name = "my_data"
    volume_type = "io1" # <<<<< Try changing this to gp2 to compare costs
    volume_size = 1000
    iops        = 800
  }
}
```

**Comment by jgrumboe** [comment]
@vdmgolub
I tried to figure out where the problem is here and searched around a little bit.
So, Macs in AWS are dedicated hosts and I used that blog https://marcincuber.medium.com/aws-mac-instances-with-terraform-701b8b292e9e to understand how Mac instances are created.
Support for creating dedicated hosts with Terraform was introduced here https://github.com/hashicorp/terraform-provider-aws/pull/10817, which gives us the new resource `aws_ec2_host`, which we would use to create a Mac host of instance family `mac1`. After creating the Mac host, you would use `aws_instance` (with `host_id` parameter) to create the `mac1.metal` instance.

AWS pricing is for the dedicated host `mac1`, for a minimum of 24h. The instance `mac1.metal` is free as you're paying for the host.

And as last note/finding: Infracost isn't supporting dedicated hosts or `aws_ec2_host` resource yet.

**Comment by vdmgolub** [comment]
@jgrumboe Thank you for researching this! 🙏 :)

**Comment by vdmgolub** [comment]
#2135 fixes this issue as well. This will go out with the next release. Thank you, @robh007, for your contribution!

**Comment by aliscott** [comment]
@injectedfusion, @caimeq, @jgrumboe this has now been released in v0.10.14.
